### PR TITLE
Add pagination to JobDetailPage

### PR DESCRIPTION
**Pull request by merobi-hub**
### Problem

We don't yet have pagination on the JobDetailPage's Run History tab for runs like we do for datasets and jobs on those routes.

### Solution

This uses the same approach employed in those cases to add pagination to the JobDetailPage. As in those cases, here, the API is queried to ensure an accurate count. But, unlike those cases, we lacked a query to get the total run count, so one has been added to RunDao. 

> **Note:** All database schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

One-line summary: adds pagination to JobDetailPage.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

Screenshot from 2023-11-06 10-43-13

### CODE REVIEW

**Comment by phixMe** [comment]
@merobi-hub We need to update the openapi spec and regenerate the docs for the doc section of the website as well here.

File: api/src/main/java/marquez/db/RunDao.java [hunk 1]
```diff
@@ -192,6 +192,9 @@ LEFT OUTER JOIN (
       """)
   List<Run> findAll(String namespace, String jobName, int limit, int offset);
 
+  @SqlQuery("SELECT count(*) FROM runs_view AS r WHERE r.job_name = :jobName")
+  int countFor(String jobName);
```
Comment (phixMe) [inline api/src/main/java/marquez/db/RunDao.java:196]: I think you need to add namespace here if I understand correctly to make the filter the same between the runs call and the count.

Comment (merobi-hub) [reply api/src/main/java/marquez/db/RunDao.java:196]: Thanks, @phixMe . When I try adding the namespace to the filter query, the API returns an error saying that the runs_view lacks a namespace column. The logs also show that the endpoint already receives the correct namespace. So is this OK as-is, or is some other change required?

Comment (phixMe) [reply api/src/main/java/marquez/db/RunDao.java:196]: You'd have to compare the runs query to the total count query... They need to be exactly the same except the count query should just have the `select count` at the beginning.

**Comment by merobi-hub** [comment]
@phixMe When I was working on this, I couldn't diagnose the issues with the API. But I'll take another look when I have a moment.

**Pull request closed without merging**